google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java

```diff
@@ -0,0 +1,451 @@
+
+    // Creates a list of remove operations.
+    for (String adGroupCriterion : adGroupCriteria) {
+      operations.add(AdGroupCriterionOperation.newBuilder().setRemove(adGroupCriterion).build());
+    }
+
+    // Creates the ad group criterion service.
+    try (AdGroupCriterionServiceClient adGroupCriterionServiceClient =
+        googleAdsClient.getLatestVersion().createAdGroupCriterionServiceClient()) {
+      // Removes the ad group criterion.
+      MutateAdGroupCriteriaResponse response =
+          adGroupCriterionServiceClient.mutateAdGroupCriteria(
+          Long.toString(customerId), operations);
+      // Gets and prints the results.
+      System.out.printf("Removed %d ad group criteria.%n", response.getResultsCount());
+      for (MutateAdGroupCriterionResult result : response.getResultsList()) {
+        System.out.printf(
+            "Successfully removed ad group criterion with resource name '%s'.%n",
+            result.getResourceName());
+      }
+    }
+  }
+
+  /**
+   * Finds all of user list ad group criteria under a campaign.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param campaignId the campaign under which to search the ad group criteria.
+   * @return a list of the ad group criteria resource names.
+   */
+  private List<String> getUserListAdGroupCriterion(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      long campaignId) {
+    List<String> userListCriteria = new ArrayList<>();
+    // Creates the Google Ads service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Creates a request that will retrieve all of the ad group criteria under a campaign.
+      SearchGoogleAdsRequest request =
+          SearchGoogleAdsRequest.newBuilder()
+              .setCustomerId(Long.toString(customerId))
+              .setPageSize(PAGE_SIZE)
+              .setQuery("SELECT ad_group_criterion.criterion_id"
+                  + " FROM ad_group_criterion"
+                  + " WHERE campaign.id = " + campaignId
+                  + " AND ad_group_criterion.type = 'USER_LIST'")
+              .build();
+      // Issues the search request.
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Iterates over all rows in all pages. Prints the results and adds the ad group criteria
+      // resource names to the list.
+      for (GoogleAdsRow googleAdsRow : searchPagedResponse.iterateAll()) {
+        String adGroupCriterionResourceName = googleAdsRow.getAdGroupCriterion().getResourceName();
+        System.out.printf(
+            "Ad group criterion with resource name '%s' was found.%n",
+            adGroupCriterionResourceName);
+        userListCriteria.add(adGroupCriterionResourceName);
+      }
+    }
+  return userListCriteria;
```

Comment:
Formatting. Maybe you want to configure Intellij to run google-java-format on all files? Then ctrl+shift+f formats the entire file (IIRC that's the default key combo).

http://screenshot/uPkTzLyGCH2